services/vod-to-live/service/src/domains/smil/models/utils.ts

```diff
@@ -0,0 +1,132 @@
+import { SECOND_IN_MILLISECONDS } from '../../../common';
+import { HeaderMetadataNames } from './header-metadata-names';
+import {
+  Event,
+  EventStream,
+  HeaderMetadata,
+  OutOfNetworkIndicator,
+  Parallel,
+  ParallelReference,
+  Reference,
+  Signal,
+  SMILEnvelope,
+  SpliceInfoSection,
+  SpliceInsert,
+} from './smil-envelope';
+
+export const createParallel = (
+  reference: ParallelReference,
+  eventStream?: EventStream | undefined,
+  clipBegin?: number | undefined,
+  clipEnd?: number | undefined,
+): Parallel => ({
+  '@clipBegin': transformSecondsToWallClock(clipBegin),
+  '@clipEnd': transformSecondsToWallClock(clipEnd),
+  audio: reference.audio,
+  video: reference.video,
+  EventStream: eventStream,
+});
+
+export const transformSecondsToWallClock = (
+  seconds?: number | undefined,
+): string | undefined => {
+  if (!seconds) {
+    return undefined;
+  }
+  const date = new Date(seconds * SECOND_IN_MILLISECONDS);
+  return `wallclock(${date.toISOString()})`;
+};
+
+export const createSmilEnvelope = (
+  parallels: Parallel[],
+  headerMetadata: HeaderMetadata[],
+): SMILEnvelope => ({
+  smil: {
+    '@xmlns': 'http://www.w3.org/2001/SMIL20/Language',
+    head: {
+      meta: headerMetadata,
+    },
+    body: {
+      seq: {
+        par: parallels,
+      },
+    },
+  },
+});
+
+export const getDefaultMetadataHeaders = (
+  vod2LiveStartTime: Date,
+): HeaderMetadata[] => {
+  return [
+    createHeaderMetadata(HeaderMetadataNames.Vod2Live, true),
+    createHeaderMetadata(
+      HeaderMetadataNames.Vod2LiveStartTime,
+      vod2LiveStartTime.toISOString(),
+    ),
+    //`splice_media` should be set to `true`
+    // but is set to `false` temporary due to bug in Origin affecting the HLS streams
+    //todo: set `splice_media` to `true`, when Origin bug is fixed
```

Comment:
Would be nice to also add a reference to the bug, e.g. a link to a github issue or similar. If it exists ofc

Reply:
The bug is internal to the origin components, so there is no public link to it. Adjusted the comment with a bit more details.